agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/localstorage/LocalFileLoader.java

```diff
@@ -95,23 +95,22 @@ PersistedFile loadTelemetriesFromDisk() {
       return null;
     }
 
-    ByteBuffer ikeyByteBuffer;
-    ByteBuffer result;
+    byte[] ikeyBytes = new byte[36];
+    int rawByteLength = (int) tempFile.length() - 36;
+    byte[] telemetryBytes = new byte[rawByteLength];
     try (FileInputStream fileInputStream = new FileInputStream(tempFile);
         FileChannel channel = fileInputStream.getChannel()) {
-      ikeyByteBuffer = ByteBuffer.allocate(36);
-      result = ByteBuffer.allocate((int) (tempFile.length() - 36));
-      channel.read(ikeyByteBuffer);
-      channel.read(result);
-      result.flip();
+      fileInputStream.read(ikeyBytes, 0, 36);
+      fileInputStream.read(telemetryBytes, 0, rawByteLength);
```

Comment:
need to check the return value of `read` here also. I'd recommend writing a small `readFully()` method that loops until it's either read the requested number of bytes, or run out of bytes